### ISSUE TITLE
collapsible_if false positive

### ISSUE THREAD

**Issue by piscisaureus** [rust-lang/rust-clippy#6593]
A `#[cfg(unix)]` attribute can make an if statement non-collapsible, but its presence is ignored by clippy.
`cargo clippy --fix` drops the attribute altogether.

## Lint name

collapsible-if

## Example

```rust
fn test() {
  if condition() {
    // ...
  } else {
    #[cfg(unix)]
    if let Some(unix_only) = resource.downcast::<UnixOnlyType>() {
      // ...
    }
  }
}
```

## Meta

```
~🦕 cargo clippy -V
clippy 0.1.51 (e38fb30 2021-01-14)

~🦕 rustc -Vv
rustc 1.51.0-nightly (e38fb306b 2021-01-14)
binary: rustc
commit-hash: e38fb306b7f5e65cca34df2dab1f0db15e1defb4
commit-date: 2021-01-14
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
LLVM version: 11.0
```
